docs/sources/operations/automatic-stream-sharding.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Automatic Stream Sharding
+menuTitle: Automatic stream sharding
+description: Automatic stream sharding can control issues around the per-stream rate limit
+weight: 110
+---
+
+# Automatic stream sharding
+
+Automatic stream sharding will attempt to keep streams under a `desired_rate` by adding new labels and values to
+existing streams. When properly tuned, this should eliminate issues where log producers are rate limited due to the
+per-stream rate limit.
+
+## The issue mitigated by automatic stream sharding
+
+Large log streams present several problems for Loki, namely increased and uneven resource usage on Ingesters and
+Distributors. The general recommendation is to explore existing log streams for additional label values that are both
+useful for querying and sufficiently low cardinality. There are many cases, however, where no more labels can
+be extracted, or cardinality for a label is dangerously large. To protect itself from such volume leading to operational failure, Loki implements per-stream rate limits;
+but the result is that some data is lost. The per-stream limit also needs human intervention to change, which is not ideal when log volumes increase and decrease.
+
+Automatic stream sharding is used to avoid large streams and rate limiting for any log stream by ensuring it is close to
+a configured `desired_rate`.
+
+## How automatic stream sharding works
+
+Automatic stream sharding works by adding a new label, `__stream_shard__`, to streams and incrementing its value to try
+and keep all streams below a configured `desired_rate`.
+
+The feature adds a new API to Ingesters that reports the size of all existing log streams. Once per second, Distributors
+query the API to get a picture of all stream rates in the system. Distributors use the existing stream-rate data and a
+configured `desired_rate` to determine how many shards a given stream should have. The desired number of new log streams
+are created with the label `__stream_shard__` and logs are divided evenly among the streams.
+
+Because automatic stream sharding is reactive and relies on successive calls to Ingesters, the view of current rates is
+always somewhat behind. As a result, the actual size of sharded streams will always be higher than the `desired_rate`.
+In practice, this is still sufficient to keep log producers from being rate limited by per-stream rate limits.
+
+## Configuration
+
+Enable automatic sharding by setting the global or per-tenant override `shard_streams`. This configuration contains
+an `enabled` flag to turn the feature on, a `desired_rate` configuration for the desired stream rate, and an
+optional `logging_enabled` flag to enable debug logging of stream sharding.
+
+*NOTE*: Setting `logging_enabled` may affect the ingestion performance of Loki.
+
+## Automatic Stream Sharding metrics
```

Comment:
All the other headings have been sentence case (only the first word capitalized), why is this one title case?

Reply:
To help clarify my comment, why is this "Automatic Stream Sharding metrics" (three words capitalized, but the feature name has NOT been capitalized anywhere else in the text) instead of "Automatic stream sharding metrics" (with only the first word capitalized, which matches other headings and text references)?

Reply:
I think it just snuk in. Do we have any docs on "how to capitalize?" It's something I consistently struggle with

Reply:
Re: Guidance on capitalization, it looks like the docs team has written up some guidance.  But feel free to ping me with questions.

https://grafana.com/docs/writers-toolkit/style-guide/capitalization-punctuation/

Reply:
Thank you!